tower-http/src/compression/body.rs

```diff
@@ -1,320 +1,233 @@
 #![allow(unused_imports)]
 
-use super::Encoding;
+use crate::compression_utils::{
+    AsyncReadBody, BodyIntoStream, DecorateAsyncRead, IoBody, WrapBody,
+};
 #[cfg(feature = "compression-br")]
 use async_compression::tokio::bufread::BrotliEncoder;
 #[cfg(feature = "compression-gzip")]
 use async_compression::tokio::bufread::GzipEncoder;
 #[cfg(feature = "compression-deflate")]
 use async_compression::tokio::bufread::ZlibEncoder;
-use bytes::{Buf, Bytes, BytesMut};
-use futures_core::Stream;
+use bytes::{Buf, Bytes};
 use futures_util::ready;
-use http::{header, HeaderValue, Response};
+use http::HeaderMap;
 use http_body::Body;
 use pin_project::pin_project;
 use std::{
-    fmt, io,
-    marker::PhantomData,
+    io,
     pin::Pin,
     task::{Context, Poll},
 };
-use tokio_util::{
-    codec::{BytesCodec, FramedRead},
-    io::StreamReader,
-};
+use tokio_util::io::StreamReader;
 
 /// Response body of [`Compression`].
 ///
 /// [`Compression`]: super::Compression
 #[pin_project]
-pub struct CompressionBody<B: Body> {
-    #[pin]
-    inner: BodyInner<B, B::Error>,
-}
+pub struct CompressionBody<B>(#[pin] pub(crate) BodyInner<B>)
```

Comment:
Why this change in struct?

Reply:
Don't exactly remember. I think I moved some things around and had to make the field `pub(crate)` to create it from another module. And making it a tuple struct made things a bit less verbose in other places.